src/scripts/components/widgets/modals/ModalStake.js

```diff
@@ -65,6 +66,10 @@ class ModalStake extends Component<Props, Object> {
     const stakeAmountWei = paratii.eth.web3.utils.toWei(String(stakeAmount))
     const videoIdStaked = this.props.selectedVideo.id
 
+    this.setState({
+      disableButton: true
```

Comment:
I think _preferably_ we would store `this.state.isStaking` instead of `disableButton` directly. This way we are storing, in `state`, the actual thing that we are tracking.

Then we can have a method in the component like so:

```javascript
disableButton = () => {
   return this.state.isStaking;
}
```

Seems a bit over-engineered, but is better, I think, in case we ever have to add other conditions as to when the button needs to be disabled